src/plugins/backend/testmod_backend.c

```diff
@@ -0,0 +1,363 @@
+/**
+ * @file
+ *
+ * @brief Tests for backend plugin
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+#include "backendprivate.h"
+#include <stdlib.h>
+#include <string.h>
+
+#include <kdbconfig.h>
+
+#include <tests_plugin.h>
+
+KeySet * set_simple (void)
+{
+	return ksNew (50, keyNew ("system/elektra/mountpoints/simple", KEY_END),
+
+		      keyNew ("system/elektra/mountpoints/simple/config", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/config/anything", KEY_VALUE, "backend", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/config/more", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/config/more/config", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/config/more/config/below", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/config/mountpoint", KEY_VALUE, "user/tests/backend/simple", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/config/path", KEY_END),
+
+		      keyNew ("system/elektra/mountpoints/simple/error", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/error/prerollback", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/error/prerollback/#1", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/error/prerollback/#1/name", KEY_VALUE, KDB_DEFAULT_STORAGE, KEY_END),
+
+		      keyNew ("system/elektra/mountpoints/simple/get", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/get/pregetstorage", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/get/pregetstorage/#0", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/get/pregetstorage/#0/config", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/get/pregetstorage/#0/config/anything", KEY_VALUE, "plugin", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/get/pregetstorage/#0/config/more", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/get/pregetstorage/#0/config/more/config", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/get/pregetstorage/#0/config/more/config/below", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/get/pregetstorage/#0/config/path", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/get/pregetstorage/#0/name", KEY_VALUE, KDB_DEFAULT_STORAGE, KEY_END),
+
+		      keyNew ("system/elektra/mountpoints/simple/set", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/set/presetstorage", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/set/presetstorage/#0", KEY_END),
+		      keyNew ("system/elektra/mountpoints/simple/set/presetstorage/#0/name", KEY_VALUE, KDB_DEFAULT_STORAGE, KEY_END),
+		      KS_END);
+}
+
+KeySet * set_default (void)
+{
+	return ksNew (30, keyNew ("default", KEY_END), keyNew ("default/config", KEY_END),
```

Comment:
None of the keys in this key set have valid names, therefore the key set will be empty.
The names either have to start with a `/` or a valid namespace, which `default` is not (yet).